### PR TITLE
[YouTube] Fix some decryption exceptions by retrying

### PR DESCRIPTION
**Pull request by Stypox**
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.

By looking at the HAR from https://github.com/TeamNewPipe/NewPipe/issues/3753#issuecomment-687558649 it seems like YouTube sometimes does not provide the player config object even though it should. In that case the extractor would crash, since it didn't have the needed information to decrypt the signatures. Therefore this PR detects when there is no player config object, and in that case runs the same request again, with at most 3 attempts. This should drastically reduce the number of exceptions caused by YouTube not providing the needed information (e.g. if the probability was 1% before, now it should be 0.0001%. Small, isn't it? ;-) ).

Closes TeamNewPipe/NewPipe#3753

### CODE REVIEW

**Review by TobiGr — COMMENTED** [review]
Looks good to me.

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java [hunk 1]
```diff
@@ -647,12 +646,31 @@ public void onFetchPage(@Nonnull Downloader downloader) throws IOException, Extr
...
+            // sometimes at random YouTube does not provide the player config,
+            // so just retry the same request three times
+            int attempts = 2;
+            while (true) {
+                playerConfig = initialAjaxJson.getObject(2).getObject("player", null);
+                if (playerConfig != null) {
+                    break;
+                }
+
+                if (attempts <= 0) {
```
[the remark this answers is not in the document]
Comment (opusforlife2) [reply extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java:663]: Why? Doesn't hurt to be careful and cover all cases.

[the remark this answers is not in the document]
Comment (TobiGr) [reply extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java:663]: Ok

**Comment by opusforlife2** [comment]
You haz a test apk?

**Comment by Stypox** [comment]
@opusforlife2 here it is: [app-debug.zip](https://github.com/TeamNewPipe/NewPipeExtractor/files/5328484/app-debug.zip)

**Comment by xibr** [comment]
I have tested this apk and no crashes yet.

I will confirm it tomorrow after further use. 

Thanks

**Comment by xibr** [comment]
I tested this apk file and it looks like the issue has been fixed, the app has not been crashing with me and is working fine.

I suggest adding the apk file and linking it to issue [#3753](https://github.com/TeamNewPipe/NewPipe/issues/3753) so that people who have the same problem can test it. 

Thanks

**Comment by Stypox** [comment]
@xibr basically everyone has this problem from time to time ;-)

**Comment by opusforlife2** [comment]
Branch name is missing in the package ID. Can the extractor branches be read by the build.gradle of the other repo?

**Comment by xibr** [comment]
@Stypox You're right, for me this apk fixed the problem.  Because I can reproduce the problem on Release version 0.20.0

**Review by TobiGr — APPROVED** [review]

**Comment by opusforlife2** [comment]
Yay!

**Pull request merged**